src/euroleague_api/team_lineup_euroleague_api.py

```diff
@@ -0,0 +1,98 @@
+    
+    # Getting pbp for the match
+    pbp = PlayByPlay()
+    pbp_data = pbp.get_game_play_by_play_data(season=season, gamecode=game_code)
+    pbp_data['Indices'] = range(0, len(pbp_data)) # resetting indices since going to loop through them, don't want any duplicates
+    pbp_data.set_index('Indices', inplace=True) # resetting indices since going to loop through them, don't want any duplicates
+
+    # Getting starting five
+    boxscoredata = BoxScoreData()
+    game_bxscr_stats = boxscoredata.get_player_boxscore_stats_data(season=season, gamecode=game_code)
+
+    starting_five = game_bxscr_stats.loc[game_bxscr_stats["IsStarter"] == 1, ["Team", "Player"]]
+    starting_five['ID'] = starting_five.groupby('Team').cumcount()
+    starting_five = starting_five.pivot(index='ID', columns='Team', values='Player')
+    starting_five.reset_index(drop=True, inplace=True)
+
+    # Creating player columns
+    on_court_cols = ['Team_A1', 'Team_A2', 'Team_A3', 'Team_A4', 'Team_A5', 'Team_B1', 'Team_B2', 'Team_B3', 'Team_B4', 'Team_B5']
+    pbp_data[on_court_cols] = np.nan
+
+    # Filling start of the match with starters
+    on_court_mask1 = ['Team_A1', 'Team_A2', 'Team_A3', 'Team_A4', 'Team_A5']
+    on_court_mask2 = ['Team_B1', 'Team_B2', 'Team_B3', 'Team_B4', 'Team_B5']
+
+    pbp_data.loc[0, on_court_mask1] = starting_five.iloc[:, 0].tolist()
+    pbp_data.loc[0, on_court_mask2] = starting_five.iloc[:, 1].tolist()
+
+    # Capturing subs and replacing player columns with players who substitute in
+    players = pbp_data.loc[min(pbp_data.index.tolist()), on_court_cols]
+
+    for idx in pbp_data.index.tolist(): # looping through plays of a game
+        if pbp_data.loc[idx, 'PLAYTYPE'] == 'IN':
+            sub_time = pbp_data.loc[idx, 'MARKERTIME']
+            sub_min = pbp_data.loc[idx, 'MINUTE']
+            sub_team = pbp_data.loc[idx, 'TEAM']
+            all_subs = pbp_data.loc[(pbp_data.MINUTE == sub_min) & (pbp_data['MARKERTIME'] == sub_time) & (pbp_data.PLAYTYPE.isin(['IN', 'OUT'])), :].sort_values(by='PLAYTYPE')
+
+            if pd.isnull(pbp_data.loc[max(all_subs.index), 'Team_A1']): # if I haven't made any replacement yet
+                subs_team1 = all_subs.loc[all_subs['TEAM'] == sub_team, :]
+                subs_team2 = all_subs.loc[all_subs['TEAM'] != sub_team, :]
+                in_players_1 = []
+                out_players_1 = []
+                for i in subs_team1.index: # capturing every sub made by the first team at that particular time in the match
+                    if subs_team1.loc[i, 'PLAYTYPE'] == 'IN':
+                        tmp_in_player = subs_team1.loc[i, 'PLAYER']
+                        in_players_1.append(tmp_in_player)
+                    elif subs_team1.loc[i, 'PLAYTYPE'] == 'OUT':
+                        tmp_out_player = subs_team1.loc[i, 'PLAYER']
+                        out_players_1.append(tmp_out_player)
+                
+                in_players_2 = []
```

Comment:
I don't go through every substitution row, that team 2 player may never come later.

The structure of the data is strange, not like NBAs where every substitution corresponds to a single row. Here, every substitution corresponds to 2 rows and they don't always appear the same way (e.g., first 'IN' row after 'OUT' row), there is no apparent pattern to it. So, I get the exact time where the substitution occurred and check if there is any more sub made at that point. Then I have them in a dataframe and go from there.

In that dataframe, it doesn't necessarily mean that the first 'IN' row's team is same as the first 'OUT' row's team. That might be why I do it like that.

Or, I didn't get what you mean which is more likely :) But I don't "loop" through every sub, I loop and when I see one I get all the subs made at that particular point. I do it this way because of the structure of the data.

Reply:
Sure, no problem. I will approve the PR and might do some more testing my side too. When I publish the package, I will ask you to test it.